python/pycylon/pycylon/data/table.pyx

```diff
@@ -1260,10 +1310,30 @@ cdef class Table:
         tb_index = self.index
         py_arrow_table = self.to_arrow().combine_chunks()
         if isinstance(key, slice):
-            new_tb = self.from_arrow(self.context, py_arrow_table.slice(key.start, key.stop))
-            new_index = tb_index.values[key.start: key.stop].tolist()
-            new_tb.set_index(new_index)
-            return new_tb
+            if key.start != None:
+                new_tb = self.from_arrow(self.context, py_arrow_table.slice(key.start, key.stop))
+                new_index = tb_index.values[key.start: key.stop].tolist()
+                new_tb.set_index(new_index)
+                return new_tb
+            elif key.start == None and key.stop > 0:
+                return self._get_head_response(0, key.stop)
+            elif key.start == None and key.stop < 0:
+                    return self._get_tail_response(0, key.stop*(-1))
```

Comment:
Thanks for adding all the options. It was missing from the initial implementation.

Reply:
Yes, the whole implementation was done in multiple commits.